cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_train_dense_default_impl.i

```diff
@@ -589,7 +589,7 @@ algorithmFpType KNNClassificationTrainBatchKernel<algorithmFpType, training::def
 
     size_t sumMid = 0;
     size_t i      = 0;
-    for (; i < sampleCount; ++i)
+    for (; i < __KDTREE_MEDIAN_RANDOM_SAMPLE_COUNT; ++i)
```

Comment:
algorithmFpType samples[__KDTREE_MEDIAN_RANDOM_SAMPLE_COUNT + 1];
    const size_t sampleCount = sizeof(samples) / sizeof(samples[0]);
    should it be __KDTREE_MEDIAN_RANDOM_SAMPLE_COUNT + 1?

Reply:
good question, see this line https://github.com/icfaust/oneDAL/blob/dev/coverity-2025_0/cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_train_dense_default_impl.i#L601  when this for loop terminates ```i == sampleCount``` (```__KDTREE_MEDIAN_RANDOM_SAMPLE_COUNT + 1```) so when it is used on that line, it will exceed the array bound.

Reply:
good point, agree with you,  but probably its a bug, check the line
https://github.com/oneapi-src/oneDAL/pull/2890/files#diff-e5b67a6ac7d03badd31a155779cec86abb98f260008cd311acf5166a0102b123R519
does it make sense to do the same fix here? My point that if we replace sampleCount var with __KDTREE_MEDIAN_RANDOM_SAMPLE_COUNT it makes sense to remove it everywhere, but probably its more correct to fix it as its done at the line 519

Reply:
What do you think about initialize sampleCount this way in this PR:
const size_t sampleCount =  __KDTREE_MEDIAN_RANDOM_SAMPLE_COUNT + 1

Reply:
done. Still don't understand why they had it as sizeof, I think @Alexandr-Solovev had a good theory.